packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/Abs_L1TokenGateway.sol

```diff
@@ -164,20 +176,28 @@ abstract contract Abs_L1TokenGateway is iOVM_L1TokenGateway, OVM_CrossDomainEnab
         );
 
         // Construct calldata for l2DepositedToken.finalizeDeposit(_to, _amount)
-        bytes memory data = abi.encodeWithSelector(
+        bytes memory message = abi.encodeWithSelector(
             iOVM_L2DepositedToken.finalizeDeposit.selector,
+            _from,
             _to,
-            _amount
+            _amount,
+            _data
         );
 
+        // Prevent tokens stranded on other side by taking
+        // the max of the user provided gas and DEFAULT_FINALIZE_WITHDRAWAL_L1_GAS
+        uint32 defaultGas = getFinalizeDepositL2Gas();
```

Comment:
I'd be tempted to simplify the bridge design and remove the default gas value altogether. Especially since our uni bridge will be used by a multitude of tokens where a common denominator for a default deposit gas will be more difficult. Plus reading memory is cheaper than reading storage unless we make the default value immutable which I don't think we want to do.

Reply:
> our uni bridge will be used by a multitude of tokens where a common denominator for a default deposit gas will be more difficult.

Very good point! Is it reasonable then to assume that front-ends can always safely estimate the required L2 gas limit for a custom L2 token?

Reply:
I agree that the gas stuff always felt a little cumbersome, definitely down to remove it.  @elenadimitrova would your proposal just be for it to forward as much gas as possible, or have a `_gas` argument to the deposit function?  I think the latter would be good.  Especially because our deposit replay function can be given higher gas if the it runs OOG the first time it's played on L2.

Reply:
I likewise prefer having a `_gas` argument. Will do that.

Reply:
Yeah down for a mandatory `_gas` argument.

Reply:
Should we mandate this to be >0 value too? In connection with discussion herehttps://github.com/ethereum-optimism/optimism/pull/824#discussion_r638445397 and here https://github.com/ethereum-optimism/optimism/pull/824#discussion_r637298290